drivers/platform/maxim/rtc_extra.h

```diff
@@ -0,0 +1,81 @@
+ *   @brief  Header file of RTC driver.
+ *   @author Ciprian Regus (ciprian.regus@analog.com)
+********************************************************************************
+ * Copyright 2022(c) Analog Devices, Inc.
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#ifndef RTC_EXTRA_H_
+#define RTC_EXTRA_H_
+
+#include <stdint.h>
+#include "no-os/rtc.h"
+
+#define MAX_RTC_MAX_VALUE	(0xFFFFF)
+
+/**
+ * @struct rtc_init_maxim
+ * @brief maxim specific init parameter
+ */
+struct rtc_init_maxim {
+	/** Load value for miliseconds */
+	uint32_t ms_load;
+};
+
+/**
+ * @enum rtc_interrupt_id
+ * @brief maxim specific interrupts
+ */
+enum rtc_interrupt_id {
+	RTC_TIMEOFDAY_INT,
+	RTC_SUBSEC_INT
+};
+
+/**
+ * @struct maxim_rtc_alarm_desc
+ * @brief maxim specific interrupt descriptor
+ */
+struct maxim_rtc_alarm_desc {
+	enum rtc_interrupt_id int_id;
+	uint32_t irq_time;
+};
+
+int32_t rtc_get_time(uint32_t *, uint32_t *);
+int32_t rtc_enable_irq(enum rtc_interrupt_id, uint32_t);
```

Comment:
For these irq/callback functions, again, I think we need to discuss as a team what to do with them, I'm not sure if this is a good way to make them public to the rest of the code.